resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventInputImpl.java

```diff
@@ -66,12 +65,21 @@ public boolean isClosed()
 
    public InboundSseEvent read() throws IOException
    {
-      byte[] chunk = null;
+      Charset charset = StandardCharsets.UTF_8;
+      if (mediaType != null && mediaType.getParameters().get(MediaType.CHARSET_PARAMETER) != null)
+      {
+         charset = Charset.forName(mediaType.getParameters().get(MediaType.CHARSET_PARAMETER));
+      }
+      if (!textLike && escape) {
+         charset = Charset.forName("ISO-8859-1");
```

Comment:
Why do you use this encoding in this case?

Reply:
This is for only for bytes event stream , like to handle this stream resource method:
```
   @GET
   @Path("get/bytes")
   @Produces(MediaType.APPLICATION_OCTET_STREAM)
   @Stream
   public Flowable<byte[]> getBytes() {
```

Reply:
Well, the bytes can be in latin1, but over the wire they must be converted to UTF8. Is that the case here?

Reply:
If this is the bytes stream like I posted above and client side expect byte type, this encoding will be used. For the text date message, all will be encoded with UTF-8 when read the event data.  The data media type controls this flag.